pvlib/tests/test_singlediode.py

```diff
@@ -110,12 +110,12 @@ def build_precise_iv_curve_dataframe(file_csv, file_json):
 
     # parse strings to np.float64
     is_array = ['Currents', 'Voltages', 'diode_voltage']
-    joined[is_array] = joined[is_array].applymap(
+    joined[is_array] = joined[is_array].map(
         lambda a: np.asarray(a, dtype=np.float64)
     )
     is_number = ['v_oc', 'i_sc', 'v_mp', 'i_mp', 'p_mp', 'i_x', 'i_xx',
                  'Temperature']
-    joined[is_number] = joined[is_number].applymap(np.float64)
+    joined[is_number] = joined[is_number].map(np.float64)
```

Comment:
Looking at the CI test failures, it seems like `DataFrame.map` only exists in newer versions of pandas.  I think this one can be replaced with just `joined[is_number'].astype(np.float64)`.  For the one on line 113, maybe just fall back to python like this?

```python
    for col in ['Currents', 'Voltages', 'diode_voltage']:
        joined[col] = [np.asarray(a, dtype=np.float64) for a in joined[col]]
```